### ISSUE TITLE
Build breaks with clang-19: implicit instantiation of undefined template 'std::char_traits<unsigned char>'

### ISSUE THREAD

**Issue by yurivict** [brainboxdotcc/DPP#1332]
**Git commit reference**
10.0.29

**Describe the bug**
```
In file included from /wrkdirs/usr/ports/www/dpp/work/DPP-10.0.29/src/dpp/cluster.cpp:21:
In file included from /usr/include/c++/v1/map:2175:
In file included from /usr/include/c++/v1/functional:552:
In file included from /usr/include/c++/v1/__functional/boyer_moore_searcher.h:27:
In file included from /usr/include/c++/v1/vector:325:
In file included from /usr/include/c++/v1/__format/formatter_bool.h:19:
In file included from /usr/include/c++/v1/__format/formatter_integral.h:21:
In file included from /usr/include/c++/v1/__format/formatter_output.h:22:
In file included from /usr/include/c++/v1/__format/parser_std_format_spec.h:39:
/usr/include/c++/v1/string:820:42: error: implicit instantiation of undefined template 'std::char_traits<unsigned char>'
  820 |   static_assert(is_same<_CharT, typename traits_type::char_type>::value,
      |                                          ^
/wrkdirs/usr/ports/www/dpp/work/DPP-10.0.29/library/../include/dpp/dispatcher.h:2016:29: note: in instantiation of template class 'std::basic_string<unsigned char>' requested here
 2016 |         std::basic_string<uint8_t> audio_data = {};
      |                                    ^
/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /wrkdirs/usr/ports/www/dpp/work/DPP-10.0.29/src/dpp/cluster.cpp:21:
In file included from /usr/include/c++/v1/map:2175:
In file included from /usr/include/c++/v1/functional:552:
In file included from /usr/include/c++/v1/__functional/boyer_moore_searcher.h:27:
In file included from /usr/include/c++/v1/vector:325:
In file included from /usr/include/c++/v1/__format/formatter_bool.h:16:
In file included from /usr/include/c++/v1/__format/concepts.h:16:
In file included from /usr/include/c++/v1/__format/format_parse_context.h:16:
/usr/include/c++/v1/string_view:300:42: error: implicit instantiation of undefined template 'std::char_traits<unsigned char>'
  300 |   static_assert(is_same<_CharT, typename traits_type::char_type>::value,
      |                                          ^
/usr/include/c++/v1/__type_traits/is_convertible.h:30:99: note: in instantiation of template class 'std::basic_string_view<unsigned char>' requested here
   30 | struct _LIBCPP_TEMPLATE_VIS is_convertible : public integral_constant<bool, __is_convertible(_T1, _T2)> {};
      |                                                                                                   ^
```

[log](https://pkg-status.freebsd.org/beefy18/data/main-amd64-default/p380be9c7980a_s5036d9652a5/logs/dpp-discord-10.0.29_1.log)

**To Reproduce**
Regular build.

**System Details:**
 - OS: FreeBSD 14.1

**Additional context**
clang-19 introduced multiple changes and this error is because of these changes in clang behavior.

**Comment by braindigitalis** [comment]
why 10.0.29, it is practically a year out of date?

**Comment by braindigitalis** [comment]
the fix for this went into 10.0.31

https://github.com/brainboxdotcc/DPP/pull/1195/files

we do not backport fixes to old versions as old versions stop working on discord API. in approximately a week's time most of these old versions are going to break due to discord deprecating one of their voice ciphers, along with lack of support for monetization and a bunch of other needed discord API features.

**Comment by yurivict** [comment]
The port wasn't updated, that's why 10.0.29

I'll reopen if it it fails again.
